src/components/account.mjs

```diff
@@ -0,0 +1,40 @@
+// @flow
+
+import crypto from 'crypto';
+
+class Account {
+  id: string;
+  balance: number;
+
+  constructor() {
+    this.id = crypto.randomBytes(32).toString('hex');
+    this.balance = 0;
+  }
+
+  async updateBalance(diff: number): Promise<number> {
+    if (this.balance + diff < 0) {
+      throw new Error("Can't spend more than your balance.");
+    }
+
+    this.balance += diff;
```

Comment:
We talked about evaluating balance from previous transactions rather than storing balance amounts and operating on those values. 
Bitcoin implements a very different but efficient approach to do that. Just for our knowledge:
https://komodoplatform.com/whats-utxo/ 
I think it would be better to understand Blockchain use cases, efficiency on the side, to use either what we talked about or UTXO.

Reply:
right, I did it like that because it was simpler but we should do it like you say, with no balance stored in each account (to easily hackable)